consul/config.sls

```diff
@@ -20,6 +20,7 @@ consul-script-install-{{ loop.index }}:
     - source: {{ script.source }}
     - name: {{ script.name }}
     - template: jinja
+    - context: {{ script.get('context', {}) | yaml }}
```

Comment:
@hugochinchilla what is the goal of `yaml` filter here? The value of `context` will be converted anyway.

I'd suggest something like this:
```
- context: {{ script.context | default({}) }}
```

Reply:
@AAbouZaid The `yaml` filter is usually necessary because of special characters (for YAML parser), which could be present in Pillar file with proper escaping. Jinja renderer would just put them in place as is.

However, using `default` looks better, but it basically does the same as `get()` method.

http://yaml.org/spec/1.1/#id868518

Reply:
@vutny Yeah using `default` filter is just a matter of style (the first part before `yaml` filter).

But I didn't get using `yaml` filter since the values should be already dumped.
Could you please give an example for that case where `yaml` is needed to get better understating?

Thanks.